server/db/schemas/course-schema.js

```diff
@@ -0,0 +1,14 @@
+import mongoose, { Schema } from "mongoose";
+import User from "./user-schema";
+
+const courseSchema = new Schema(
+    {
+        course_id: String,
+        number: String,
+        dept: String,
+        prof: User.schema,
```

Comment:
this should just be a ID to a user to save space and avoid having the whole User embedded in